amundsen_application/static/js/utils/navigation-utils.ts

```diff
@@ -1,22 +1,32 @@
 import * as qs from 'simple-query-string';
 import { createBrowserHistory } from 'history';
 
+import { ResourceType } from 'interfaces/Resources';
+
 // https://github.com/ReactTraining/react-router/issues/3972#issuecomment-264805667
 export const BrowserHistory = createBrowserHistory();
 
 interface SearchParams {
   term?: string;
-  resource?: string;
+  resource?: ResourceType;
   index?: number;
+  filters?: {};
 }
 
 export const updateSearchUrl = (searchParams: SearchParams, replace: boolean = false) => {
-  // Explicitly listing out parameters to ensure consistent URL format
-  const urlParams = qs.stringify({
+  const filtersForResource = searchParams.filters && searchParams.filters[searchParams.resource] || {};
+  const queryStringValues = {
     term: searchParams.term,
     resource: searchParams.resource,
     index: searchParams.index,
-  });
+  };
```

Comment:
`filters: undefined` will get dropped by `qs` while `filters: {}` will result in `filters=%7B%7D`. Maybe we can leverage that to simplify?

The comment about explicitly listing out parameters should apply to line 18.

Reply:
I've moved the comment but left the logic as is. `filters` is coming from the the application state and it is an object and when there are no filters it will be an empty dictionary. Because of that I don't see a way of simplifying this piece of logic unless we allow the `filters` state type to be an object or undefined...which would lead to having to check for that in different areas of the code.